otter/convergence/service.py

```diff
@@ -799,3 +808,18 @@ def divergent_changed(self, children):
             # the return value is ignored, but we return this for testing
             eff = self._converge_all(my_buckets, children)
             return perform(self._dispatcher, self._with_conv_runid(eff))
+
+
+class ConvergenceExecutor(object):
+    gather = attr.ib()
+    plan = attr.ib()
+    get_desired_group_state = attr.ib()
+    convergence_succeeded = attr.ib()
```

Comment:
Yes! I agree, and like I said in the original PR comment:
`The exact methods contained in ConvergenceExecutor may change. For example, just UpdateServersCache instead of the entirety of convergence_succeeded_servers since most of the code there isn't specific to launch_server.`